server/src/main/java/org/elasticsearch/index/seqno/RetentionLeaseBackgroundSyncAction.java

```diff
@@ -0,0 +1,185 @@
+ */
+public class RetentionLeaseBackgroundSyncAction extends TransportReplicationAction<
+        RetentionLeaseBackgroundSyncAction.Request,
+        RetentionLeaseBackgroundSyncAction.Request,
+        ReplicationResponse> {
+
+    public static String ACTION_NAME = "indices:admin/seq_no/retention_lease_background_sync";
+
+    private static final Logger LOGGER = LogManager.getLogger(RetentionLeaseSyncAction.class);
+
+    protected Logger getLogger() {
+        return LOGGER;
+    }
+
+    @Inject
+    public RetentionLeaseBackgroundSyncAction(
+            final Settings settings,
+            final TransportService transportService,
+            final ClusterService clusterService,
+            final IndicesService indicesService,
+            final ThreadPool threadPool,
+            final ShardStateAction shardStateAction,
+            final ActionFilters actionFilters,
+            final IndexNameExpressionResolver indexNameExpressionResolver) {
+        super(
+                settings,
+                ACTION_NAME,
+                transportService,
+                clusterService,
+                indicesService,
+                threadPool,
+                shardStateAction,
+                actionFilters,
+                indexNameExpressionResolver,
+                Request::new,
+                Request::new,
+                ThreadPool.Names.MANAGEMENT);
+    }
+
+    /**
+     * Background sync the specified retention leases for the specified shard.
+     *
+     * @param shardId         the shard to sync
+     * @param retentionLeases the retention leases to sync
+     */
+    public void backgroundSync(
+            final ShardId shardId,
+            final RetentionLeases retentionLeases) {
+        Objects.requireNonNull(shardId);
+        Objects.requireNonNull(retentionLeases);
+        final ThreadContext threadContext = threadPool.getThreadContext();
+        try (ThreadContext.StoredContext ignore = threadContext.stashContext()) {
+            // we have to execute under the system context so that if security is enabled the sync is authorized
+            threadContext.markAsSystemContext();
+            execute(
+                    new Request(shardId, retentionLeases),
+                    ActionListener.wrap(
+                            r -> {},
+                            e -> {
+                                if (ExceptionsHelper.unwrap(e, AlreadyClosedException.class, IndexShardClosedException.class) == null) {
+                                    getLogger().warn(new ParameterizedMessage("{} retention lease background sync failed", shardId), e);
+                                }
+                            }));
+        }
+    }
+
+    @Override
+    protected PrimaryResult<Request, ReplicationResponse> shardOperationOnPrimary(final Request request, final IndexShard primary) {
+        Objects.requireNonNull(request);
+        Objects.requireNonNull(primary);
+        primary.afterWriteOperation();
```

Comment:
I am not sure if this call is useful for it does not change translog nor Lucene.